### ISSUE TITLE
Cannot select target migration in update database, drop down list always contails only one option: 0

### ISSUE THREAD

**Issue by vivi00790** [JetBrains/rider-efcore#218]
#### Describe the bug

As title, see screenshot.

**To Reproduce**

1. Add a migration through "Add Migration" prompt.
2. Open "Update Database" prompt.
3. Click on "Target migration" drop-down.
4. See an error

#### Expected behavior

Base on screenshot, I except it would show 0 and Initial.

#### Screenshots

![image](https://github.com/JetBrains/rider-efcore/assets/6245453/dc8eb2da-f978-49a0-a1e5-690d90c949bb)

### Environment (please complete the following information):

- OS: Win 10
- .NET SDK info (installed ones and version used within Rider): 8.0.204 (both)
- Plugin version: bundled 241.14494.307
- Rider version: 2024.1 build #241.14494.307

**Comment by DmitryZhelnin** [comment]
I have the same issue

**Comment by jektvik** [comment]
+1

**Comment by jektvik** [comment]
No progress since April. Jetbrains please 'unblock' -> it used to be possible to type in the migration name manually so a part of the issue is the decision to hardcode the list to take dropdown entries only.

**Comment by seclerp** [comment]
Hi folks! This issue requires detailed repro steps as I can't reproduce it using a regular EF Core project setup (neither in a single project setup nor with a separate startup + migrations one)

@jektvik your case sounds interesting, I assume, that it may be a key for the problem resolution. Could you please describe your project's anatomy in a little more detail? 

Do you have references configured between projects like the following?

```mermaid
graph TD;
    A[Startup project]-->B;
    B[Migrations project]-->C[DbContext project];
```

**Comment by jektvik** [comment]
I have 2 projects:
1. project with migrations - cmd line project, startup capable incl connection string
2. project with db context (and main app/models code), in my case an azure functions project

project dependency is: migrations csproj references dbcontext csproj

**Comment by seclerp** [comment]
And which one you usually select as Startup project and which one as Migrations project in the Update Database dialog?

**Comment by vivi00790** [comment]
@seclerp My case is the same as @jektvik 's.
As you can see in the screenshot, my startup and migration project are both set to the project containing migration.
Plus startup project only contains 1 option which is the project containing migration, there's no other options I can select.

**Comment by lukhipolito** [comment]
I have the same issue.
there are 3 projects involved:
A is where DbContext is
B is where migrations are
C is the startup project

C depends on B which depends on A

**Comment by seclerp** [comment]
Hi @JucianoSilva153, the plugin got bundled into Rider in 2023, which means there is no need to release new separate plugin releases. Please don't be confused by older releases, they are kept for compatibility reasons for very old Rider versions for users with fallback licenses. 

.NET 9 is fully supported right now too, I will update the README to reflect that.

**Comment by JucianoSilva153** [comment]
@seclerp  Greetings, and thanks for the clarification!

I really thought the plugin wasn't compatible with .NET 9. As I mentioned in my previous comment, I had multiple projects targeting .NET 9, and in my Persistence project, the plugin wasn't able to detect the startup project (the API project). However, after changing the target framework from .NET 9 to .NET 8, everything worked perfectly! So that i thought it was the problem!!

Sorry for the confusion, and thank you again for the explanation!

**Comment by seclerp** [comment]
@JucianoSilva153 support for .NET 9 projects in EF Core plugin was added in some of the Rider 2024.3 minor release, in Rider 2025.1 it should work fine too. In case it still doesn't work, I would kindly ask you to create a separate issue here or on our YouTrack. Thanks!

**Comment by samusaran** [comment]
I've run into this issue today after updating Rider to 2025.1.

Is there any update?

**Comment by Lyncee59** [comment]
Resolved!

Installed 
- Microsoft.EntityFrameworkCore.Design
- Microsoft.EntityFrameworkCore.Tools

Error disappeared and can select target framework for whoever pass by !

**Comment by vivi00790** [comment]
> Resolved!
> 
> Installed
> 
> * Microsoft.EntityFrameworkCore.Design
> * Microsoft.EntityFrameworkCore.Tools
> 
> Error disappeared and can select target framework for whoever pass by !

Thank you for solving my 1.5 year confuse!
Tho I'm fired form that job and not using EF at current position lol

**Comment by seclerp** [comment]
Wow, really nice to hear that the issue has finally been resolved for you folks!

Please mind the requirements covered [in the README](https://github.com/JetBrains/rider-efcore?tab=readme-ov-file#requirements) for further use:
> TL;DR:
> - ...
> - officially supported versions of .NET, EF Core  tools **and EF Core NuGets**.

And
> Before opening the issue, please make sure that your projects and development environment completely satisfies these requirements:
> - ...
> - **Microsoft.EntityFrameworkCore.*: 5.0.0 or higher**

Maybe it needs better and more specific wording. I will think about it.
